src/plugins/saml-auth/components/AuthorizationPopup.jsx

```diff
@@ -0,0 +1,155 @@
+      // unset url search
+      this.close()
+      window.history.pushState({}, document.title, window.location.pathname)
+    } else if(isSamlFailed) {
+      window.history.pushState({}, document.title, window.location.pathname)
+    }
+  }
+
+  render() {
+    let {
+      samlAuthSelectors,
+      authSelectors,
+      authActions,
+      getComponent,
+      errSelectors,
+      specSelectors,
+      getSystem,
+      fn: { AST = {} },
+    } = this.props
+    let definitions = authSelectors.shownDefinitions()
+    let authorized = authSelectors.authorized()
+
+    let Auths = getComponent("auths")
+    let loginDisclaimer = specSelectors.spec().get("loginDisclaimer")
+
+    let { selectedDefinitionOption } = this.state
+    let errors = errSelectors
+      .allErrors()
+      .filter((err) => err.get("authId") === "SamlAuth")
+    let hasErrors = errors.size > 0
+    let isAuthenticated = authorized.size > 0
+    let authenticatedKey = authorized.keySeq().first()
+    let selectedDefinitionKey = isAuthenticated ? authenticatedKey : selectedDefinitionOption
+
+    let isSamlAuthenticating =
+      samlAuthSelectors.samlAuthState() === SAML_AUTH_STATE_LOGGING_IN
+    let showLoginOptions = !isSamlAuthenticating && !selectedDefinitionKey
+    let showLoginStep =
+      !isSamlAuthenticating &&
+      definitions &&
+      !!selectedDefinitionKey
+
+    return (
+      <div className="dialog-ux">
+        <div className="backdrop-ux"></div>
+        <div className="modal-ux">
+          <div className="modal-dialog-ux">
+            <div className="modal-ux-inner">
+              <div className="modal-ux-header">
+                <h3>Authorization</h3>
+                <button
+                  type="button"
+                  className="close-modal"
+                  onClick={this.close}
+                >
+                  <svg width="20" height="20">
+                    <use href="#close" xlinkHref="#close" />
+                  </svg>
+                </button>
+              </div>
+              <div className="modal-ux-content">
+
+                {isSamlAuthenticating && (
+                  <div className="saml-auth-info">Authenticating SAML Token...</div>
+                )}
+                {showLoginOptions && (
+                  <DefinitionSelect
+                    definitions={definitions}
+                    onSelect={this.onSelectDefinition}
+                    getComponent={getComponent}
+                  />
+                )}
+                {hasErrors &&
+                  errors.map((error) => (
+                    <div key={""} className="login-error">
+                      {error.get("message")}
+                    </div>
+                  ))}
+                {showLoginStep &&
+                  definitions
+                    .filter((definition) => {
+                      const [key] = definition.keys()
+                      return key === selectedDefinitionKey
+                    })
```

Comment:
rename `key`? This is confusing with the `key` used in the `.map()` which is an index